packages/code-du-travail-frontend/src/conventions/ConventionTexte.js

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { fetchTexte } from "../common/convention.service";
+import Sidebar from "./texte/Sidebar";
+import Content from "./texte/Content";
+import styled from "styled-components";
+import { theme } from "@cdt/ui/";
+
+class ConventionTexte extends React.Component {
+  constructor(props) {
+    super(props);
+    if (props.preloadedTexte) {
+      const texte = props.preloadedTexte;
+      const topNode = this.getFirstNodeWithChildren(texte);
+      this.state = { topNode, texte, loaded: true };
+    } else {
+      this.state = { loaded: false };
+    }
+  }
+
+  async componentDidMount() {
+    if (this.state.texte) {
+      return false;
+    }
+    const { id } = this.props;
+    const texte = await fetchTexte({ id });
+    const topNode = this.getFirstNodeWithChildren(texte);
+    this.setState({ texte, topNode, loaded: true });
+  }
+
+  getFirstNodeWithChildren(texte) {
+    let topNode = texte;
+    while (topNode.children && topNode.children.length == 1) {
+      topNode = topNode.children[0];
+    }
+    return topNode.children ? topNode : texte;
+  }
+
+  onChangeSummaryTitleExpanded(expanded, sectionId) {
```

Comment:
you can use this notation too 
```
onChangeSummaryTitleExpanded = (sectionId, expanded) => {
}
```

so you can directly use it in render 
```
<Sidebar
              topNode={topNode}
              onSummaryTitleToggleExpanded={this.onChangeSummaryTitleExpanded}
            />
```

Reply:
thanks!